lib/engine/action/base.rb

```diff
@@ -58,7 +58,7 @@ def to_h
           'created_at' => @created_at.to_i,
           'auto_actions' => @auto_actions.empty? ? nil : @auto_actions.map(&:to_h),
           **args_to_h,
-        }.reject { |_, v| v.nil? }
```

Comment:
is this right? i think the intent was to explicitly check for nil, it’s not necessarily true you want to get rid of false here

Reply:
It's true that someone could introduce an action param which defaults to true. However, that's a bad idea and I think it's fine not to support it. (Better to invert the semantics of the param to default to false.)

Just to be sure, I scanned through the entire database dump I received. The only params to appear with value false are program params: auto_pass_after, from_market, indefinite, skip_buy, skip_track, skip_close, unconditional. These are all prepared to handle nil/absent values. (Values like auto_pass_after already have to deal with nil at runtime because they were extensions to an already existing program.)

Filtering false values has the same benefits as filtering nil, e.g. smaller payloads, less boilerplate in the representation, fewer diffs in fixtures when adding/changing/removing fields.

Reply:
i don't think it's safe to do this, let's get rid of this change

Reply:
Okay, reverting.